src/core/tokenizer.c

```diff
@@ -195,3 +197,78 @@ tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
 	free(content);
 	return NULL;
 }
+
+
+
+/*
+ * tokenize: split the text into tokens separated by a whitespace
+ * and pass the token through the filter pipeline.
+ */
+tokenset_t *
+tokenize2(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
+{
+
+	UBreakIterator *it_token;
+	UChar          utext[text_len + 1]; 
+	UErrorCode     ec = U_ZERO_ERROR;
+	int32_t end;
+	int32_t start;
+	char *content;
+	tokenset_t *tset;
+	char val[10];
+
+	if ((tset = tokenset_create()) == NULL) {
+		return NULL;
+	}
+	if ((content = strdup(text)) == NULL) {
+		goto err;
+	}
+
+	u_uastrcpy(utext, text);
+	it_token = ubrk_open(UBRK_WORD, "en_us", utext, u_strlen(utext), &ec);
```

Comment:
We can still implement more rules ICU rules.
https://unicode-org.github.io/icu/userguide/boundaryanalysis/break-rules.html